### DISCUSSION TITLE
How can I use Jolt physics from my GDExtention?

### DISCUSSION THREAD

**Discussion by William-Godwin** [godot-jolt/godot-jolt#821] [Questions] [answered]
I don’t quite understand the mechanism for working with Jolt through Godot extensions. If I call the physics server functions in GDExtention will I work with the built-in Godot physics? And if so, how can I use Jolt physics from my extension?

I'm completely new to C++ so... Sorry for the stupid question. =)

**Comment by mihe** [comment, marked as answer]
> If I call the physics server functions in GDExtention will I work with the built-in Godot physics?

No, it'll use Jolt. Godot Physics isn't "loaded" at all if you've selected `JoltPhysics3D` as your physics engine, so anything physics-based will use Jolt.

With that said, you won't be able to use certain parts of this extension from statically compiled languages like C++, like the substitute joints (`JoltHingeJoint3D`, etc.) without jumping through annoying hoops, but those are largely optional anyway. See #632 for more details on that.

**Comment by William-Godwin** [comment]
Okay! Thanks a lot for your answer! And so fast =)

Interesting situation. Have you thought about making a version with a module? How did the Mirror? (off topic)

**Reply by mihe** [reply]
> Have you thought about making a version with a module?

It's looking likely that this extension will be ported over to be an official Godot module in a not too distant future.

> How did the Mirror?

The Mirror has its own Jolt implementation, which is implemented as a module.
